packages/react/src/components/Table/TableBody/TableBodyRowRenderer.jsx

```diff
@@ -198,7 +198,7 @@ const TableBodyRowRenderer = (props) => {
         data-testid={`${tableId}-lazy-row-${row.id}`}
       >
         {hasRowSelection === 'multi' ? <TableCell /> : null}
-        {hasRowExpansion ? <TableCell /> : null}
+        {hasRowExpansion || hasRowNesting ? <TableCell /> : null}
```

Comment:
We can use the `totalColumns` which contains the number of visible columns prop (calculated by the table and passed down to its subcomponents) and simplify this section by only writing:  

```
  {totalColumns.map((v, colIndex) => (
    <TableCell key={`empty-cell-${colIndex}`}>
      <SkeletonText />
    </TableCell>
  ))}
```

Reply:
I had to add a fix to make `hasRowNesting` add an extra column to the totalColumn count, but otherwise, it's much cleaner. Thanks!

Reply:
Man I'm sorry. But I was probably wrong here. This would work fine if we were just inserting empty cells for the whole row, but I realise now that only the actual data columns should show skeleton texts. Here all extra spacer columns also get the skeleton:

![image](https://user-images.githubusercontent.com/5167091/151047753-a4745cec-d537-4b82-aad7-7fcedcf618f7.png)

Your original solution was the best option here

Reply:
reverted.